src/app/delegation_backend/src/itn_uptime_analyzer/time.go

```diff
@@ -0,0 +1,121 @@
+package itn_uptime_analyzer
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	sheets "google.golang.org/api/sheets/v4"
+)
+
+// Returns current time in UTC format
+func GetCurrentTime() time.Time {
+	currentTime := time.Now().UTC()
+	return currentTime
+}
+
+// Checks the spreadsheet for a possible value for the last execution date
+// If there is no date, or the date is between the +/- one hour range of the chosen execution interval then the function returns current time minus the execution interval
+func GetLastExecutionTime(config AppConfig, client *sheets.Service, log *logging.ZapEventLogger, sheetTitle string, currentTime time.Time, executionInterval int) time.Time {
+	readRange := fmt.Sprintf("%s!A%d:Z%d", sheetTitle, 1, 1)
+	spId := config.AnalyzerOutputGsheetId
+
+	resp, err := client.Spreadsheets.Values.Get(spId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v\n", err)
+	}
+
+	var lastFilledColumn int = len(resp.Values[0]) - 1
+	var lastExecutionBasedOnSheetsAsTime time.Time
+
+	readRange = fmt.Sprintf("%s!%s%d", sheetTitle, string(lastFilledColumn+65), 1)
```

Comment:
Use `strconv` package to do conversion. This one is buggy, and there's a warning by Go for it (at least from my go1.20 environment).

Reply:
In fact: 
```suggestion
	readRange = fmt.Sprintf("%s!%d%d", sheetTitle, lastFilledColumn+65, 1)
```

Reply:
May I ask for a comment on mysterious role of `65` number? It's completely unclear to me.

Reply:
If it's max number of columns, maybe it makes more sense to have a constant like `MAX_COLUMNS` and use it here  and in other occurrences of `65`?

Reply:
There are other occurrences of wrong `string(<int>)` usage, please make sure they're fixed

Reply:
65 is the ASCII number for the letter A and to find out what is the letter of a column we find the index for said column and add the ASCII number of A to it. The casting `string(<int>)` is done so that we would get the corresponding letter to <int>.

@georgeee